### PR TITLE
[13.0] [MIG] agreement -> base_contract

### PR DESCRIPTION
**Pull request by i-vyshnevska**
migrate agreement

### CODE REVIEW

**Comment by max3903** [comment]
@i-vyshnevska Please have a look at #386

**Comment by gurneyalex** [comment]
@i-vyshnevska  can you rename the module base_contract?

**Review by TDu — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Do we finally have an agreement - hehe - on the renaming?

**Review by sbejaoui — CHANGES_REQUESTED** [review]
Please consider the discussion about 13.0 road-map #386

**Review by jgrandguillaume — APPROVED** [review]
Work well on functional side

File: base_contract/models/agreement.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    partner_id = fields.Many2one(
+        "res.partner",
+        string="Partner",
+        ondelete="restrict",
+        domain=[("parent_id", "=", False)],
+    )
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        default=lambda self: self.env["res.company"]._company_default_get(),
```
Comment (guewen) [inline base_contract/models/agreement.py]: `_company_default_get()` should be removed: https://github.com/odoo/odoo/blob/58492733b81073694a7e2f5e790b31983f91df7d/odoo/addons/base/models/res_company.py#L205

It shows a warning at loading of the module.

```suggestion
        default=lambda self: self.env.company.id,
```

File: base_contract/views/agreement.xml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                    class="oe_stat_button" icon="fa-archive">
+                    <field name="active" widget="boolean_button"
+                        options='{"terminology": "archive"}'/>
+                </button>
+            </div>
+            <group name="main">
+                <group name="left">
+                    <field name="agreement_type_id"
+                           groups="agreement.group_use_agreement_type"/>
+                    <field name="is_template" groups="agreement.group_use_agreement_template"/>
```
Comment (guewen) [inline base_contract/views/agreement.xml]: Warnings:

```
2020-05-06 13:44:29,775 115 WARNING odoodb odoo.addons.base.models.ir_ui_view: The group agreement.group_use_agreement_type defined in view agreement.form does not exist! 
2020-05-06 13:44:29,776 115 WARNING odoodb odoo.addons.base.models.ir_ui_view: The group agreement.group_use_agreement_template defined in view agreement.form does not exist! 
```

```suggestion
                    <field name="agreement_type_id"
                           groups="base_contract.group_use_agreement_type"/>
                    <field name="is_template" groups="base_contract.group_use_agreement_template"/>
```

I didn't check for other places

**Comment by alexis-via** [comment]
I fixed and improved several points in this PR via a PR on @i-vyshnevska 's branch https://github.com/i-vyshnevska/contract/pull/1

**Comment by tbaden** [comment]
What's the progress on this one? Is there still something todo or does it just need some more reviews?

**Review by alexis-via — APPROVED** [review]

**Review by Chabagutor — APPROVED** [review]
Tested and installed for prodution purpose.

**Comment by alexis-via** [comment]
This is ready to be merged. @sbejaoui 's comment is one year old and he asks to take into account the roadmap discussion, and it's precisely what has been implemented here. So let's move on and merge.

**Review by sbejaoui — APPROVED** [review]

**Comment by alexis-via** [comment]
There is something I don't understand with "pre-commit run -a" : when I run it on this branch, it doesn't change anything... although Travis complains about black/isort stuff.
If someone with more experience than me on this can sort it out... it's the last thing needed to merge.

**Comment by rvalyi** [comment]
@alexis-via probably you should try to rebase this branch on 13.0 before trying your local precommit.
Because from Travis logs it complains about XML formatting from the Prettier tool. But in this 13-mig-agreement, the precommit configuration `.pre-commit-config.yaml` (but may bve also some of the other dot files) is not up to date like in the 13 branch, specially this branch version has no such prettier formating instruction that's why it wont even try to run locally. Instead on Travis it looks it's running the 13.0 pre-commit configuration.

compare:
https://github.com/i-vyshnevska/contract/blob/13-mig-agreement/.pre-commit-config.yaml
https://github.com/OCA/contract/blob/13.0/.pre-commit-config.yaml#L24

Is it clearer now?

**Comment by alexis-via** [comment]
A few days ago, I created another PR to replace this one: https://github.com/OCA/contract/pull/578

**Comment by newtratip** [comment]
@i-vyshnevska please close this pr supersede https://github.com/OCA/contract/pull/628

**Pull request closed without merging**